Ewald.js

```diff
@@ -0,0 +1,120 @@
+// voegt een submit event handler toe aan het form
+document.getElementById('ibanForm').addEventListener('submit', genereerIBAN);
+
+// Hiermee kan je string de bank code ophalen
+var bankIdentificatie = [
+        ['ABNAMRO', 'ABNA'],
+        ['INGBANK', 'INGB'],
+        ['RABOBANK', 'RABO'],
+        ['LANSCHOT', 'FVLB'],
+        ['SNSBANK', 'SNSB'],
+        ['DEMIR', 'DHBN']
+];
+
+function genereerIBAN(oEvent) {
+    oEvent.preventDefault();
+
+    // Een IBAN bestaat uit een aantal onderdelen:
+    // (1) een Landcode, bestaande uit twee letters. We doen in deze opdracht alleen Nederlandse rekeningen,
+    //    dus je kunt 'NL' gebruiken.
+    // (2) een controlegetal bestaande uit twee cijfers.
+    // (3) de rekeningidentificatie
+    //    Dit onderdeel bestaat uit het relevante deel uit de globale variabele 'bankIdentificatie' en het rekeningnummer.
+    //    Het rekeningnummer moet 10 tekens lang zijn. Is het korter, zet er dan nullen voor.
+    var landCode = 'NL';
+    var rekeningNummer = document.getElementById('rekeningnummer').value;
+    var bankSelect = document.getElementById('bankSelect');
+    var gekozenBank = bankSelect.options[bankSelect.selectedIndex].value;
+
+    var bankCode           = bepaalBankCode(gekozenBank);
+    rekeningNummer         = bepaalHeleRekeningNummer(rekeningNummer);
+
+    var controleGetal      = bepaalControleGetal(bankCode, landCode, rekeningNummer);
+
+    document.getElementById("iban").innerHTML = 'Je IBAN is: ' + landCode + controleGetal + bankCode + rekeningNummer;
+}
+
+// de bankCode kun je halen uit de globale 'bankIdentificatie' array.
+// We zouden de bankCode natuurlijk ook gewoon in de <option>s van de <select> kunnen plaatsen..
+// maar dan hoef je niet te oefenen met arrays :)
+function bepaalBankCode(bank) {
+    var bankCode = '';
+    for(var i=0; i < bankIdentificatie.length; i++){
+    	if(bankIdentificatie[i][0]===bank){
+				bankCode = bankIdentificatie[i][1];
+      }
+    }
+    // haal hier de bankCode op uit 'bankIdentificatie'
+
+    return bankCode;
+}
+
+// Maak van het 7- of 9-cijferige nummer een 10-cijferig rekeningnummer
+function bepaalHeleRekeningNummer(rekeningNummer) {
+
+    // zorg er hier voor dat het rekeningnummer altijd 10 cijfers lang is, eventueel met nullen ervoor
+    while(rekeningNummer.length < 10){
+      rekeningNummer = '0'+rekeningNummer;
+    }
+
+    while(rekeningNummer.length > 10){

```

Comment:
Netjes dat je ook de mogelijkheid om meer dan 10 tekens in te vullen afvangt!
Maar met slice(0,10) reduceer je het direct tot 10 tekens, dus een while-loop is overkill. Alleen If was voldoende, en zelfs dat is niet nodig, omdat slice(0,10) gewoon de oorspronkelijke string retourneert als die 10 of minder tekens zou bevatten.